client/a8c-for-agencies/hooks/use-subscription-details.ts

```diff
@@ -0,0 +1,39 @@
+import { useLocale } from '@automattic/i18n-utils';
+import { useMemo } from 'react';
+import useFetchClientProducts from 'calypso/a8c-for-agencies/data/client/use-fetch-client-products';
+import { formatDate } from 'calypso/dashboard/utils/datetime';
+import { useSelector } from 'calypso/state';
+import { getUserBillingType } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import type { ReferralProduct } from '../sections/client/types';
+
+export const useSubscriptionDetails = ( subscription?: ReferralProduct ) => {
```

Comment:
This is similar to the one added in [#105539](https://github.com/Automattic/wp-calypso/pull/105539) but it looks like here we're only really using `expiryDate` and the other bits are unused. 
Is the idea to later consolidate the two hooks into one?